regcomp.c

```diff
@@ -24850,6 +24878,383 @@ S_parse_uniprop_string(pTHX_
+
+            /* Truncate back to the prefix, which is unvarying */
+            SvCUR_set(syllable_name, hangul_prefix_len);
+
+            sv_catpv(syllable_name, JamoL[L - LBase]);
+            sv_catpv(syllable_name, JamoV[V - VBase]);
+            sv_catpv(syllable_name, JamoT[T - TBase]);
+
+            if (execute_wildcard(subpattern_re,
+                        SvPVX(syllable_name),
+                        SvEND(syllable_name),
+                        SvPVX(syllable_name), 0,
+                        syllable_name,
+                        0))
+            {
+                *prop_definition = add_cp_to_invlist(*prop_definition,
+                                                    i + SBase);
+                found_matches = TRUE;
+            }
+        }
+    }
+
+    SvREFCNT_dec(syllable_name);
+
+    /* The rest of the algorithmically generatable names are of the form
+     * "PREFIX-code_point".  The prefixes and the code point limits of each
+     * were returned to us in the array 'algorithmic_names' from data in
+     * lib/unicore/Name.pm.  'code_point' in the name is expressed in hex. */
+    for (i = 0; i <= av_top_index((AV *) algorithmic_names); i++) {
+        IV j;
+
+        /* Each element of the array is a hash, giving the details for the
+         * series of names it covers.  There is the base name of the characters
+         * in the series, and the low and high code points in the series.  And,
+         * for optimization purposes a string containing all the legal
+         * characters that could possibly be in a name in this series. */
+        HV * this_series = (HV *) SvRV(* av_fetch((AV *) algorithmic_names, i, 0));
+        SV * prefix = * hv_fetchs(this_series, "name", 0);
+        IV low = SvIV(* hv_fetchs(this_series, "low", 0));
+        IV high = SvIV(* hv_fetchs(this_series, "high", 0));
+        char * legal = SvPVX(* hv_fetchs(this_series, "legal", 0));
+
+        /* Pre-allocate an SV with enough space */
+        SV * algo_name = Perl_newSVpvf(aTHX_ "%s-0000", SvPVX(prefix));
+        if (high >= 0x10000) {
+            sv_catpvs(algo_name, "0");
+        }
+
+        /* This series can be skipped entirely if the pattern requires
+         * something longer than any name in the series, or can only match an
+         * empty name, or contains a character not found in any name in the
+         * series */
+        if (    prog->minlen <= (SSize_t) SvCUR(algo_name)
+            &&  prog->maxlen > 0
+            && ( ! must
+                || SvCUR(must) == 0
+                || strspn(SvPVX(must), legal) == SvCUR(must)))
+        {
+            for (j = low; j <= high; j++) { /* For each code point in the series */
+
+                /* Get its name, and see if it matches the subpattern */
+                Perl_sv_setpvf(aTHX_ algo_name, "%s-%X", SvPVX(prefix),
+                                     (unsigned) j);
+
+                if (execute_wildcard(subpattern_re,
```

Comment:
This can die, which would leak anything you haven't yet refcnt_dec'd.

Reply:
I'll mortalize it